component/module/string/string.go

```diff
@@ -122,6 +124,8 @@ func (c *Component) Update(args component.Arguments) error {
 				// Go ahead and fill in the value
 				c.values[name] = val
 				c.exportComponents[cc.Name] = cc
+			case *unix.Component:
```

Comment:
If there's a clever way to do this without copying `component/prometheus/exporter/exporter.go` into `component/prometheus/exporter/unix/exporter.go` let me know. Considering how soon it sounds like we will end up changing this to not be a singleton it may not be worth a ton of time.